graylog2-server/src/test/java/org/graylog/testing/elasticsearch/ElasticsearchInstance.java

```diff
@@ -0,0 +1,162 @@
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.google.common.collect.Iterators.toArray;
+
+/**
+ * This rule starts an Elasticsearch instance and provides a configured {@link JestClient}.
+ */
+public class ElasticsearchInstance extends ExternalResource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchInstance.class);
+
+    private static final Map<Version, ElasticsearchContainer> containersByVersion = new HashMap<>();
+
+    private static final String PROPERTIES_RESOURCE_NAME = "elasticsearch.properties";
+
+    private static final String DEFAULT_IMAGE_OSS = "docker.elastic.co/elasticsearch/elasticsearch-oss";
+    private static final String DEFAULT_IMAGE = "elasticsearch";
+    private static final String DEFAULT_VERSION = "6.5.2";
+
+    private final ElasticsearchContainer container;
+    private JestClient jestClient;
+    private Client client;
+    private FixtureImporter fixtureImporter = new FixtureImporter();
+    private final Version version;
+
+    public static ElasticsearchInstance create() {
+        final Properties properties = PropertyLoader.loadProperties(PROPERTIES_RESOURCE_NAME);
+
+        return create(properties.getProperty("version", DEFAULT_VERSION));
+    }
+
+    public static ElasticsearchInstance create(String versionString) {
+        final Version version = Version.valueOf(versionString);
+        final String image = imageNameFrom(version);
+
+        LOG.debug("Creating instance {}", image);
+
+        return new ElasticsearchInstance(image, version);
+    }
+
+    private static String imageNameFrom(Version version) {
+        final String defaultImage = version.satisfies("^6.0.0")
+                // The OSS image only exists for 6.0.0 and later
+                ? DEFAULT_IMAGE_OSS
+                // For older versions do not use the official image because it contains x-pack stuff we don't want
+                : DEFAULT_IMAGE;
+        return defaultImage + ":" + version.toString();
+    }
+
+    private ElasticsearchInstance(String image, Version version) {
+        this.version = version;
+        this.container = createContainer(image, version);
+    }
+
+    private static ElasticsearchContainer createContainer(String image, Version version) {
+        if (!containersByVersion.containsKey(version)) {
+            containersByVersion.put(version, startNewContainerInstance(image));
+        }
+        return containersByVersion.get(version);
+    }
+
+    private static ElasticsearchContainer startNewContainerInstance(String image) {
+        final ElasticsearchContainer container = new ElasticsearchContainer(image)
+                .withEnv("ES_JAVA_OPTS", "-Xms512m -Xmx512m")
+                .withEnv("discovery.type", "single-node")
```

Comment:
Does this actually work as an environment variable? I only used the `discovery.type` and `action.auto_create_index` settings as program arguments via `./bin/elasticsearch -E action.auto_create_index`.

Reply:
As of the docs [here](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docker.html#docker-configuration-methods) it does